pkg/controller/backupentry/actuator.go

```diff
@@ -0,0 +1,132 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupentry
+
+import (
+	"context"
+	"strings"
+
+	controller "github.com/gardener/gardener-extensions/pkg/controller"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+)
+
+const (
+	// EtcdBackupSecretName is the name of secret having credentials for etcd backups.
+	EtcdBackupSecretName string = "etcd-backup"
+
+	backupBucketName string = "bucketName"
+)
+
+// Actuator acts upon BackupEntry resources.
+type Actuator interface {
+	// Reconcile reconciles the BackupEntry.
+	Reconcile(context.Context, *extensionsv1alpha1.BackupEntry) error
+	// Delete deletes the BackupEntry.
+	Delete(context.Context, *extensionsv1alpha1.BackupEntry) error
+}
+
+// ProviderActuator preforms provider specific operation with BackupBucket resources.
+type ProviderActuator interface {
+	// Delete deletes the BackupBucket.
+	Delete(context.Context, *extensionsv1alpha1.BackupEntry) error
+	// GetETCDSecretData return the updated the secret data as per provider requirement.
+	GetETCDSecretData(context.Context, *extensionsv1alpha1.BackupEntry, map[string][]byte) (map[string][]byte, error)
+}
+
+type actuator struct {
+	providerActuator ProviderActuator
+	client           client.Client
+	logger           logr.Logger
+}
+
+// InjectClient injects the given client into the valuesProvider.
+func (a *actuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+// InjectFunc enables injecting Kubernetes dependencies into actuator's dependencies.
+func (a *actuator) InjectFunc(f inject.Func) error {
+	return f(a.providerActuator)
+}
+
+// NewActuator creates a new Actuator that updates the status of the handled BackupBucket resources.
+func NewActuator(provideActuator ProviderActuator, logger logr.Logger) Actuator {
+	return &actuator{
+		logger:           logger,
+		providerActuator: provideActuator,
+	}
+}
+
+// Reconcile reconciles the update of a BackupEntry
+func (a *actuator) Reconcile(ctx context.Context, be *extensionsv1alpha1.BackupEntry) error {
+	return a.deployEtcdBackupSecret(ctx, be)
+}
+
+func (a *actuator) deployEtcdBackupSecret(ctx context.Context, be *extensionsv1alpha1.BackupEntry) error {
+	shootTechnicalID, _ := extractShootDetailsFromBackupEntryName(be.Name)
+	namespace := &corev1.Namespace{}
+	if err := a.client.Get(ctx, kutil.Key(shootTechnicalID), namespace); err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
Is this a case we really need to handle this way? Can you explain why we should not return an error in this situation?

Reply:
No. This is not erroneous case. Once shoot is deleted and backupEntry reconciliation is happening in deletionGracePeriod. Then seedNamespace won't be there and we don't want to deploy etcd backup Secret.

Also, at the start of shoot reconciliation, we deploy backupEntry resource independent of deploySeedNamesapce, So it might possible the seedNamespace is not created yet.